simulator/CMakeLists.txt

```diff
@@ -0,0 +1,120 @@
+#cmake
+#Building the scalar MIPS CPU simulator
+#Soshin Konstantin
+#soshinkonstantinv@gmail.com
+
+cmake_minimum_required(VERSION 2.6)
+project(mipt-mips)
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wpedantic -Wold-style-cast -Werror")
```

Comment:
Does it enable optimizations (`-O3`) and native architecture (`-mnative`) somewhere? Simulation speed decreased from 700 kHz to 40 kHz.

Reply:
No, I forgot about that. Honestly, I dont fully understand how this part of options.mk work
![image](https://user-images.githubusercontent.com/26014259/36991553-1c47acfe-20b9-11e8-8d3d-19d04fd28646.png)
(I dont know how to paste into commentary source text). Where are these macroses defined?

Reply:
Use ` ```<your code>``` `

Reply:
Makefile support several build options:
1. Default: all optimizations are enabled (`-O3 -march=native -flto`)
2. Release: optimizations are enabled, but target architecture is not set (`-O3 -flto`). It prevent compiler from emitting Intel/AMD extension instructions (SSE/AVX etc.) available on machine, so binary becomes portable.
3. Debug: all optimizations are disabled, debugging symbols are collected (`-O0 -g`)
4. Profiling: debugging and profiling symbols are collected (`-pg -g`)

Actually you may keep only 2 at the moment: Release and Debug. I believe CMake has some interfaces to set different compile flags

`OBJ_DIR` is a path to object files — they are different for different build options and compilers. With CMake we may ignore that, leaving responsibility to choose directory to user (he/she may build release version in `build` and debug version in `debugbuild`, for instance). 

Msys part is required to build MIPT-MIPS under Windows in Msys/Cygwin/MinGW environment. Not sure if it is useful anymore (I use Bash for Windows on my local machines, but AppVeyor runs on MinGW).